odoo/models.py

```diff
@@ -4888,14 +4888,19 @@ def _search(self, domain, offset=0, limit=None, order=None, count=False, access_
         query.limit = limit
 
         if count:
-            # Ignore order, limit and offset when just counting, they don't make sense and could
+            # Ignore order and offset when just counting, they don't make sense and could
             # hurt performance
-            query_str, params = query.select(limit and "1" or "count(*)")
             if limit:
-                query_str = "select count(*) from (" + query_str + ") t"
+                # Special case to avoid counting every record in DB (which can be really slow)
+                # Then the result of this case should be always between [0, limit]
```

Comment:
```suggestion
                # Special case to avoid counting every record in DB (which can be really slow).
                # The result will be between 0 and limit.
```